### ISSUE TITLE
Saving to .md files in the editor removes trailing spaces (changing markdown syntax unintentionally)

### ISSUE THREAD

**Issue by Alzurana** [godotengine/godot#96440]
### Tested versions

- Reproducible in 4.3 editor
- Other versions not tested

### System information

Windows 10 - Godot 4.3 - Forward+

### Issue description

The godot editor recognizes .md as a file format. I can open such files as text files in and crate a new file of that type. It's part of the list of possible text formats. I edited my readme.md directly in the godot editor and saved it. I then noticed that it removed trailing spaces, which are used to denote newlines in markdown. Godot shouldn't sanitize these upon saving a .md as that effectively changes the syntax

### Steps to reproduce

- Make an empty project
- Create a new text file with the .md file ending
- Write lines with double spaces at the end for line breaks
- Save, the double spaces get "cleaned off"

### Minimal reproduction project (MRP)

[markdown-example.zip](https://github.com/user-attachments/files/16829937/markdown-example.zip)

**Comment by Alzurana** [comment]
While investigating I found that trimming are editor settings:
https://github.com/godotengine/godot/blob/61598c5c88d95b96811d386cb20d714c35f4c6d7/editor/plugins/script_editor_plugin.cpp#L2942

This opens up the question if it should be configurable which file formats exept from these settings as they are generally very useful for any other format.

I would propose a new setting that comes with these 3, that determines file types exempt from adulteration/trimming. It would be a single setting that exepts all 3 if a file format is defined there.

**Comment by akien-mga** [comment]
> Godot shouldn't sanitize these upon saving a .md as that effectively changes the syntax

Does Markdown really use trailing whitespace as syntax? That's such a bad design :(
Tested it and it seems to be used for manual line breaks indeed... I tend to prefer an explicit `<br>` so it's clear.

For the record, most IDEs and text editors I've used have options to trim whitespace on save, and I haven't ever seen it specific to only some types of files.

I think excluding Markdown files specifically would probably solve the issue in your case, and be a bother for other users who don't rely on Markdown's double trailing spaces to make line breaks.

We could add another option to allow configuring which file types should be exempted from this feature but an option to configure an option starts becoming pretty niche and something that only people who somehow land on this issue would discover IMO.

**Comment by Alzurana** [comment]
> That's such a bad design :(
I agree, I never liked it. It seems to be controversial as well: https://www.markdownguide.org/basic-syntax/#line-break-best-practices
<br><br>
Trailing whitespace removal is not the default setting anyways. I had to enable it, I guess that's why few stumbled across this.
Maybe it is enough to just add to the config description that enabling this can trim newlines in markdown.

**Comment by aaronfranke** [comment]
Note that you don't actually need trailing spaces in Markdown. While `\s\s\n` (space, space, newline) is a line break in Markdown, so is `\\n` (backslash at the end of the line). You may wish to switch over to this style in your Markdown files to prevent tools (even outside Godot) from automatically trimming trailing space characters.
